main.go

```diff
@@ -2,11 +2,25 @@ package main
 
 import (
 	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
 
 	"github.com/jmorganca/ollama/cmd"
 	"github.com/spf13/cobra"
 )
 
 func main() {
+	sigChan := make(chan os.Signal, 1)
```

Comment:
This leaks the progress spinner implementation into the caller and can still leave the cursor hidden. Manipulating the cursor directly isn't ideal since users may have different cursor styles so the experience will not be uniform